test/wdio.conf.ts

```diff
@@ -307,8 +309,18 @@ export const config: Options.Testrunner = {
    * @param {Boolean} result.passed    true if test has passed, otherwise false
    * @param {Object}  result.retries   informations to spec related retries, e.g. `{ attempts: 0, limit: 0 }`
    */
-  // afterTest: function(test, context, { error, result, duration, passed, retries }) {
-  // },
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  afterTest: async (test, __, { passed }) => {
+    if (!passed) {
+      const screenshotDir = join(__dirname, 'screenshots');
+      console.log(`Test run failed. Saving a screenshot of the failure here: ${screenshotDir}`);
+      if (!existsSync(screenshotDir)) {
+        mkdirSync(screenshotDir, { recursive: true });
+      }
+      const sanitizedTestTitle = test.title.replace(/[^a-zA-Z0-9 ]/g, "");
```

Comment:
For the sanitization, we have tests which have a `-` character in them, so this character should be added.  the RegEx should probably also keep/not strip out `'` and `"` as well.